pkg/cloud/azure/services/config/startupscript.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure"
+	"sigs.k8s.io/cluster-api-provider-azure/pkg/cloud/azure/actuators"
+)
+
+// GetVMStartupScript returns startup script based on role
+func GetVMStartupScript(machine *actuators.MachineScope, bootstrapToken string) (string, error) {
+	var startupScript string
+
+	if !machine.Scope.ClusterConfig.CAKeyPair.HasCertAndKey() {
+		return "", errors.New("failed to run controlplane, missing CACertificate")
+	}
+
+	if machine.Scope.ClusterConfig.PublicIPName == "" {
+		return "", errors.New("failed to run controlplane, PublicIP name not available")
+	}
+
+	dnsName := fmt.Sprintf("%s.%s.cloudapp.azure.com", strings.ToLower(machine.Scope.ClusterConfig.PublicIPName), strings.ToLower(machine.Scope.ClusterConfig.Location))
```

Comment:
Same comment about default suffixing of `cloudapp.azure.com`. We are able to pull the FQDN from the PIP, if it exists.

Reply:
moved to defaults for now, will address when dnszone is configurable